beavis-ci.sh

```diff
@@ -0,0 +1,199 @@
+    outputformat="HTML"
+    ext="html"
+    branch="html"
+else
+    echo "Rendering the master branch notebooks:"
+    outputformat="notebook"
+    ext="nbconvert.ipynb"
+    branch="rendered"
+fi
+mkdir -p log
+webdir="https://github.com/LSSTScienceCollaborations/DMStackClub/tree/${branch}"
+
+# Some notebooks may have whitespace in their names - fix this first:
+for notebook in */*.ipynb; do
+    ipynbfile="$( echo "$notebook" | sed s/' '/'_'/g )"
+    if [ $ipynbfile != "$notebook" ]; then
+        mv -f "$notebook" $ipynbfile
+    fi
+done
+
+# Which notebooks to run? Either all of them, or the ones listed in README.rst:
+if [ $all -eq 0 ]; then
+    notebooks="$( grep ipynb README.rst |& grep -v "No such file" | grep -v rendered | grep -v html | cut -d"<" -f2 | cut -d">" -f1 | sed s/%20/'_'/g )" 
+fi
+# If README.rst is not present or empty of ipynb instances, default to running everything:
+if [ $all -gt 0 ] || [ ${#notebooks[0]} -eq 0 ]; then
+    notebooks="$( ls */*.ipynb )"
+fi
+echo "$notebooks"
+
+# Now loop over notebooks, running them one by one:
+declare -a outputs
+for notebook in $notebooks; do
+
+    # Which folder is this notebook in?
+    folder=$( dirname $notebook )
+    filename=$( basename $notebook )
+
+    # Rename files to make them easier to work with:
+    stem=$PWD/$folder/log/$filename
+    logfile=${stem%.*}.log
+    svgfile=${stem%.*}.svg
+    mkdir -p $PWD/$folder/log
+
+    # Check the notebook exists:
+    if [ -e "$notebook" ]; then
+        echo "Running nbconvert on $notebook ..."    
+    else
+        echo "WARNING: could not find $notebook, setting build status to unknown."
+        echo "$notebook: No such file in master branch" > $logfile
+        cp ../../.badges/unknown.svg $svgfile
+        continue
+    fi
+
+    # Run the notebook:
+    /opt/rh/rh-python36/root/usr/bin/jupyter nbconvert \
+                      --ExecutePreprocessor.timeout=600 --to $outputformat \
+                      --execute $notebook &> $logfile
+    
+    # Set the build status according to the output:
+    output=${notebook%.*}.$ext
+
+    if [ -e $output ]; then
+        outputs+=( $output )
+        echo "SUCCESS: $output produced."
+        cp ../../.badges/passing.svg $svgfile
+    else
+        echo "WARNING: $output was not created, read the log in $logfile for details."
+        cp ../../.badges/failing.svg $svgfile
+    fi
+    
+done
+
+if [ $no_push -gt 0 ]; then
+    sleep 0
+
+else
+    echo "Attempting to push the rendered outputs to GitHub in an orphan branch..."
+
+    git branch -D $branch >& /dev/null
+    git checkout --orphan $branch
+    git rm -rf .
+    git add -f "${outputs[@]}"
+    git add -f */log
+    git commit -m "pushed rendered notebooks and log files"
+    git push -q -f \
+     https://${GITHUB_USERNAME}:${GITHUB_API_KEY}@github.com/LSSTScienceCollaborations/DMStackClub  $branch
+    echo "Done!"
+    git checkout master
```

Comment:
Do we really want to checkout `master`? What if the user was in another branch? I think this should return the user to the branch they were in, right?